vendor/processors/stripe_processor.py

```diff
@@ -198,6 +268,197 @@ def build_subscription(self, subscription, payment_method_id):
+            price_data = self.build_price(offer, offer.current_price_object())
+            coupon_data = self.build_coupon(offer, offer.current_price_object())
+            if not stripe_product_id:
+                product = self.stripe_create_object(self.stripe.Product, product_data)
+                if meta.get('stripe', None):
+                    meta['stripe']['product_id'] = product['id']
+                else:
+                    meta['stripe'] = {'product_id': product['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_product = self.stripe_update_object(self.stripe.Product, stripe_product_id, product_data)
+                if existing_product:
+                    if meta.get('stripe', None):
+                        meta['stripe']['product_id'] = existing_product['id']
+                    else:
+                        meta['stripe'] = {'product_id': existing_product['id']}
+
+            # Check if meta has this offer stripe price and add if not
+            if not stripe_price_id:
+                price = self.stripe_create_object(self.stripe.Price, price_data)
+                if meta.get('stripe', None):
+                    meta['stripe']['price_id'] = price['id']
+                else:
+                    meta['stripe'] = {'price_id': price['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_price = self.stripe_update_object(self.stripe.Price, stripe_price_id, price_data)
+                if existing_price:
+                    if meta.get('stripe', None):
+                        meta['stripe']['price_id'] = existing_price['id']
+                    else:
+                        meta['stripe'] = {'price_id': existing_price['id']}
+
+            # Check if meta has this offer stripe coupon and add if not
+            if not stripe_coupon_id:
+                coupon = self.stripe_create_object(self.stripe.Coupon, coupon_data)
+                if meta.get('stripe', None):
+                    meta['stripe']['coupon_id'] = coupon['id']
+                else:
+                    meta['stripe'] = {'coupon_id': coupon['id']}
+            else:
+                # if it does, sync the vendor version
+                existing_coupon = self.stripe_update_object(self.stripe.Coupon, stripe_coupon_id, coupon_data)
+                if existing_coupon:
+                    if meta.get('stripe', None):
+                        meta['stripe']['coupon_id'] = existing_coupon['id']
+                    else:
+                        meta['stripe'] = {'coupon_id': existing_coupon['id']}
+
+            offer.save()
+
+    def sync_stripe_vendor_objects(self):
+        """
+        Sync up vendor customers on this site who have stripe customer objs created and ones that dont
+        Sync up all the offers on the current site to products on stripe
+        """
+        self.sync_customers()
+        self.sync_offers()
```

Comment:
Probably going to need to add the site as a parameter

Reply:
Dont need any site param as I can just call `self.site` since processor called with site object